### PR TITLE
Feature/auth as package

### PR DESCRIPTION
**Pull request by ralkov**

### CODE REVIEW

File: examples/auth/auth/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        id: uuid.v4()
+      },
+      'SECRETJWT'
+    ) // TODO - hide jwt Secret
+  },
+  loginCallback: async (_, username) => {
+    return jwt.sign(
+      {
+        name: username,
+        id: uuid.v4()
```
Comment (IhostVlad) [inline examples/auth/auth/index.js]: Identity even after success login becomes new guid random value, is it expected behavior?

File: examples/hacker-news/common/read-models/graphql/projection.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
+// @flow
```
Comment (IhostVlad) [inline examples/hacker-news/common/read-models/graphql/projection.js]: Since read-models are now based on free-style resolvers, and mandatory `graphql` is dropped, so maybe rename this read-model to more neutral naming, e.g. `default`
Most likely that `hacker-news` application is not actual. Maybe merge with dev?

File: examples/auth/auth/index.js (hunk 1 above)
Comment (ralkov) [reply examples/auth/auth/index.js]: May be in this case id is not needed. I'll remove it, thx.

File: examples/hacker-news/common/read-models/graphql/projection.js (hunk 2 above)
Comment (ralkov) [reply examples/hacker-news/common/read-models/graphql/projection.js]: done

File: packages/resolve-scripts/configs/schema.resolve.config.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "maxAge": {
+          "type": "integer"
+        }
+      },
+      "additionalProperties": false
```
Comment (IhostVlad) [inline packages/resolve-scripts/configs/schema.resolve.config.json]: Field `jwtCookie` validation is meant to have only two fields: `name` and `maxAge`, correct? But `applyJwtCookie` meant that developer can provide custom cookie options. Which is correct?

**Comment by TatyanaRyzh** [comment]
1. Return the true path to `static` folder

File: examples/hacker-news/client/middlewares/index.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  USER_LOGOUT,
+  OPTIMISTIC_STORY_UPVOTED,
+  OPTIMISTIC_STORY_UNVOTED
+} from '../actions/actionTypes'
+import { rootDirectory } from '../constants'
+import { actionTypes } from 'resolve-redux'
+const { SEND_COMMAND } = actionTypes
+
+const logoutMiddleware = () => next => action => {
```
Comment (IhostVlad) [inline examples/hacker-news/client/middlewares/index.js]: Is `logoutMiddleware` actual since now it's performed by `resolve-auth` via special HTML form?

File: examples/hacker-news/client/store/index.js [hunk 5]
```diff
@@ -1,76 +1 @@
...
-        logoutMiddleware,
-        storyCreateMiddleware,
-        optimisticVotingMiddleware
-      ]
-    : []
-
-  const enhancer = composeEnhancers(applyMiddleware(...middleware))
-  return createStore(reducer, Immutable(initialState), enhancer)
-}
+export default () => {}
```
Comment (IhostVlad) [inline examples/hacker-news/client/store/index.js]: Is this file required since it contains just empty function?

File: examples/hacker-news/package.json [hunk 6]
```diff
@@ -53,14 +54,11 @@
...
-    "resolve-bus-memory": "0.7.4",
-    "resolve-command": "0.7.4",
-    "resolve-es": "0.7.4",
-    "resolve-query": "0.7.4",
-    "resolve-readmodel-memory": "0.7.4",
     "resolve-redux": "0.7.4",
     "resolve-scripts": "0.7.4",
+    "resolve-es": "0.7.4",
     "resolve-storage-lite": "0.7.4",
+    "resolve-bus-memory": "0.7.4",
```
Comment (IhostVlad) [inline examples/hacker-news/package.json]: Does new `resolve-scripts` packages-based architecture require implicit `bus` and `es` packges enumeration in appropriate `package.json` ?

File: examples/hacker-news/storage.json [hunk 7]
```diff
@@ -1,4 +0,0 @@
-{"type":"StoryCreated","payload":{"title":"Welcome to reSolve Hacker News!","text":"To import news from the origin (https://news.ycombinator.com/), run `npm run import` or `yarn run import` and restart the application","link":"","userId":"f6118edd-aeca-46a7-99fa-56ca72555543","userName":"resolve"},"aggregateId":"0b8aa1db-85ea-46e6-a795-597bd506494e","aggregateVersion":1,"timestamp":1522141781312,"_id":"EoY818SOcOAceAWL"}
```
Comment (IhostVlad) [inline examples/hacker-news/storage.json]: This file had been stored in repository not accidentally - it contains event with information about `import` launching. It's was necessary to prevent user from disambiguation when he launches fresh downloaded example HackerNews

File: packages/resolve-auth/src/helpers.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  query: expressReq.query,
+  cookies: expressReq.cookies,
+  resolve: expressReq.resolve
+})
+
+const createResponse = expressRes => {
+  const response = {
+    expressRes,
+    clearCookie: name => {
+      const fieldContentRegExp = /^[\u0009\u0020-\u007e\u0080-\u00ff]+$/ // eslint-disable-line
```
Comment (IhostVlad) [inline packages/resolve-auth/src/helpers.js]: Is it expected behavior that RegExp checks specified *codepoints* instead of *octets* ? (See www.utf8-chartable.de/unicode-utf8-table.pl?start=0&number=1024) ES engine works with UTF-16 set internally, and suppose Unicode codepoints in regexp with `\uXXXX` literals. It does not match potential underlying bytes in appropriate HTTP header, and also does not affect MIME encoding.

File: packages/resolve-redux/package.json [hunk 9]
```diff
@@ -22,37 +22,33 @@
...
-    "apollo-cache-inmemory": "^1.1.0",
-    "apollo-client": "^2.0.2",
-    "apollo-link-http": "^1.1.0",
+    "apollo-cache-inmemory": "^1.1.9",
+    "apollo-client": "^2.2.5",
+    "apollo-link-http": "^1.5.2",
     "diff-json": "^0.1.11",
-    "graphql": "^0.11.7",
-    "graphql-tag": "^2.5.0",
+    "graphql": "^0.13.1",
```
Comment (IhostVlad) [inline packages/resolve-redux/package.json]: Is `graphql` still actual? I know most likely it had been remained wrongly after read-modules updating. Also with `apollo`.

File: packages/resolve-scripts/configs/babelrc.json [hunk 10]
```diff
@@ -0,0 +1,11 @@
+{
+  "presets": ["env", "react"],
+  "plugins": [
+    "add-module-exports",
```
Comment (IhostVlad) [inline packages/resolve-scripts/configs/babelrc.json]: Is legacy CommonJS module requirement style still being actual in 2018 year? 
CommonJS legacy support had been introduced here:
https://github.com/reimagined/resolve/commit/c8de8a4efe6f87582a09d798220c114d9aa140d5
And automated by appropriate babel plugin here:
https://github.com/reimagined/resolve/commit/41276f95eeef9436c0dcb053d7c1a204575186a9

File: packages/resolve-scripts/src/runtime/client/index.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+import Routes from './components/Routes'
+import createStore from './store/create_store'
+
+const routes = require($resolve.routes)
+
+const history = createHistory({
+  basename: process.env.ROOT_PATH
+})
+const store = createStore(window.__INITIAL_STATE__, history)
```
Comment (IhostVlad) [inline packages/resolve-scripts/src/runtime/client/index.js]: Does `InitialState` exist anymore? Nor read-models or view-models uses some *default* initial state, nor application have some *default* state without relation to some read or view model

File: packages/resolve-scripts/src/runtime/server/get_html_markup.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    `<!doctype html>` +
+    `<html ${helmet.htmlAttributes.toString()}>` +
+    '<head>' +
+    `${helmet.title.toString()}` +
+    `${helmet.meta.toString()}` +
+    `${helmet.link.toString()}` +
+    `${helmet.style.toString()}` +
+    styleTags +
+    '<script>' +
+    `window.__INITIAL_STATE__=${jsonUtfStringify(initialState)};` +
```
Comment (IhostVlad) [inline packages/resolve-scripts/src/runtime/server/get_html_markup.js]: Also as comment above - even if provide initial state to application, no view model can use it by current architecture. From another side, it can be used by custom UI reducers, e.g. for provide optimistic page loading... It's discussion queston

File: examples/hacker-news/client/store/index.js (hunk 5 above)
Comment (ralkov) [reply examples/hacker-news/client/store/index.js]: Yes. It'll be removed after fix in resolve-scripts
